### PR TITLE
Add Drop Data Source command

### PR DESCRIPTION
**Pull request by OvidiuCode**
Stable release checklist:  

* [ ] Updated the version using `python -m dev set-version {version}` 
* [x] Updated `/docs/changelog.md` with the changes for the release

Pre-release checklist:

* [ ] Updated the **Unreleased** section of `/docs/changelog.md` with the new changes.

### CODE REVIEW

File: sema4ai/src/sema4ai_code/robocorp_language_server.py [hunk 1]
```diff
@@ -2348,11 +2348,56 @@ def _import_zip_as_action_package(
...
+            http_password=password,
+        )
+
+        name = datasource.get("name")
+        engine = datasource["engine"]
+        created_table = datasource.get("created_table")
+        model_name = datasource.get("model_name")
+
+        query = ""
+        if engine == "files":
```
Comment (fabioz) [inline sema4ai/src/sema4ai_code/robocorp_language_server.py:2375]: I think you should verify if what they want to drop exists and return a "success" result with a message saying that nothing happened because it wasn't there (it's strange to receive a message saying that it was successfully dropped when it didn't even exist).

-- p.s.: don't do this now, let's merge and do it in a separate PR.

File: sema4ai/vscode-client/src/extension.ts [hunk 2]
```diff
@@ -1003,3 +1018,69 @@ export async function getLanguageServerPythonInfo(): Promise<InterpreterInfo | u
...
+        }
+        let selectedItem = await showSelectOneQuickPick(captions, "Please select which Data Source to drop.");
+
+        if (!selectedItem) {
+            return;
+        }
+
+        datasource = selectedItem.action;
+    } else {
+        datasource = entry?.extraData?.datasource;
```
Comment (fabioz) [inline sema4ai/vscode-client/src/extension.ts:1072]: In this use case (when it's clicked in the view), I think we should confirm if the user really wants to do it -- in the other when it's already selected in a drop then this is not needed.

File: sema4ai/vscode-client/src/extension.ts [hunk 3]
```diff
@@ -1003,3 +1018,69 @@ export async function getLanguageServerPythonInfo(): Promise<InterpreterInfo | u
...
+        "datasource": datasource,
+        "data_server_info": dataServerStatus["data"],
+    });
+
+    if (!result["success"]) {
+        window.showErrorMessage(result["message"] || "Unknown error");
+        return;
+    }
+
+    window.showInformationMessage("Data Source dropped successfully.");
```
Comment (fabioz) [inline sema4ai/vscode-client/src/extension.ts:1085]: I think that in the success case, the `result['message']` should be shown (so, we can say that either it was dropped successfully or it wasn't dropped because it didn't exist -- that's not an error, so, it's a 'success' with a 'message').

File: sema4ai/vscode-client/src/extension.ts [hunk 4]
```diff
@@ -1003,3 +1018,69 @@ export async function getLanguageServerPythonInfo(): Promise<InterpreterInfo | u
...
+
+        let captions: QuickPickItemWithAction[] = new Array();
+        for (const source of dataSources) {
+            let caption: QuickPickItemWithAction = {
+                "label": getDataSourceCaption(source),
+                "action": source,
+            };
+            captions.push(caption);
+        }
+        let selectedItem = await showSelectOneQuickPick(captions, "Please select which Data Source to drop.");
```
Comment (fabioz) [inline sema4ai/vscode-client/src/extension.ts:1064]: It'd be nice to have a `Drop All Data Sources from the Action Package` too and add a drop button in the parent of the data sources to drop all.

-- p.s.: don't do this now, let's merge and do it in a separate PR.

File: sema4ai/src/sema4ai_code/robocorp_language_server.py [hunk 5]
```diff
@@ -2348,11 +2348,56 @@ def _import_zip_as_action_package(
...
+            query = f"DROP TABLE files.{created_table}dsada"
+
+        elif name == "custom" and created_table:
+            query = f"DROP TABLE {name}.{created_table}"
+
+        elif engine.startswith("prediction:") or (name == "custom" and model_name):
+            query = f"DROP MODEL IF EXISTS {name}.{datasource.get('model_name')}"
```
Comment (fabioz) [inline sema4ai/src/sema4ai_code/robocorp_language_server.py:2382]: Use the `model_name` variable that's already in the scope... could probably return an success=False if `model_name` is not given.

File: sema4ai/src/sema4ai_code/robocorp_language_server.py [hunk 6]
```diff
@@ -2348,11 +2348,56 @@ def _import_zip_as_action_package(
...
+        )
+
+        name = datasource.get("name")
+        engine = datasource["engine"]
+        created_table = datasource.get("created_table")
+        model_name = datasource.get("model_name")
+
+        query = ""
+        if engine == "files":
+            query = f"DROP TABLE files.{created_table}dsada"
```
Comment (fabioz) [inline sema4ai/src/sema4ai_code/robocorp_language_server.py:2376]: Should probably verify that `created_table` was provided and return success=False if it's not.

File: sema4ai/src/sema4ai_code/robocorp_language_server.py [hunk 7]
```diff
@@ -2348,11 +2348,56 @@ def _import_zip_as_action_package(
...
+            query = f"DROP MODEL IF EXISTS {name}.{datasource.get('model_name')}"
+        else:
+            query = f"DROP DATABASE IF EXISTS {name}"
+
+        try:
+            connection.run_sql(query)
+        except Exception as e:
+            log.exception(f"Error while executing query: {query}. {str(e)}")
+            return ActionResult.make_failure(
+                f"Error while dropping data source."
```
Comment (fabioz) [inline sema4ai/src/sema4ai_code/robocorp_language_server.py:2391]: Should probably add the error when dropping the data source.

i.e.: `f"Error while dropping data source: {e}"`

**Review by fabioz — CHANGES_REQUESTED** [review]

File: sema4ai/vscode-client/src/viewsRobots.ts [hunk 8]
```diff
@@ -780,6 +793,14 @@ export class RobotsTreeDataProvider implements vscode.TreeDataProvider<RobotEntr
                 "arguments": [element.robot.directory],
             };
             treeItem.collapsibleState = vscode.TreeItemCollapsibleState.None;
+        } else if (element.type === RobotEntryType.DropDataSource) {
```
[the remark this answers is not in the document]
Comment (OvidiuCode) [reply sema4ai/vscode-client/src/viewsRobots.ts:796]: Right, we don't need it as we will probably never add it to a specific Commands section.

File: sema4ai/src/sema4ai_code/robocorp_language_server.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (OvidiuCode) [reply sema4ai/src/sema4ai_code/robocorp_language_server.py:2376]: good catch!

File: sema4ai/src/sema4ai_code/robocorp_language_server.py (hunk 5 above)
Comment (OvidiuCode) [reply sema4ai/src/sema4ai_code/robocorp_language_server.py:2382]: I would just leave it fail when dropping

**Review by fabioz — APPROVED** [review]
LGTM
